sdk/src/signing.rs

```diff
@@ -544,15 +544,15 @@ pub async fn wrap_tx<'a, N: Namada<'a>>(
                 let target = namada_core::types::masp::TransferTarget::Address(
                     fee_payer_address.clone(),
                 );
-                let fee_amount = DenominatedAmount {
+                let fee_amount = DenominatedAmount::new(
```

Comment:
This variable is only used to feed the next call to `gen_shielded_transfer`. This second function though, takes a `DenominatedAmount` but only ever uses the `amount` field of it. I was wondering if we could maybe change the signature of `gen_shielded_transfer` to take an `Amount` instead